### PR TITLE
feat: ucan invocation post handled car

### PR DESCRIPTION
**Pull request by vasco-santos**
Adds POST of handled CAR invocations to w3infra's https://github.com/web3-storage/w3infra/pull/171

Implementation details:
- Performed in `waitUntil`. We have no way to rollback previous operations if it fails. We should have a failure tolerance here with CF Queue. Let's create an issue and sort this out in follow up.
- Only makes sure in tests that endpoint returns 200. We need to discuss on how we should perform integration tests for this!

Note that `fetchMock` was not working with current version of `miniflare`. Upgrading miniflare creates issues with D1 which I was not able to solve easily. The issues looked to be with `undici` version behind current miniflare release. Upgraded in the lock file manually just `undici` and is now working! 

Needs:
- [x] secret set

### CODE REVIEW

File: packages/access-api/src/bindings.d.ts [hunk 1]
```diff
@@ -71,6 +72,8 @@ export interface RouteContext {
     validations: Validations
   }
   uploadApi: ConnectionView
+  uploadApiUrl: URL
```
Comment (alanshaw) [inline packages/access-api/src/bindings.d.ts:75]: I know these are one and the same but we should confiugure it has a separate service here.

```suggestion
  ucanInvocationPostURL: URL
```

File: packages/access-api/src/routes/raw.js [hunk 2]
```diff
@@ -17,9 +21,24 @@ export async function postRaw(request, env) {
...
 
+  const body = new Uint8Array(await request.arrayBuffer())
   const rsp = await server.request({
-    body: new Uint8Array(await request.arrayBuffer()),
+    body,
     headers: Object.fromEntries(request.headers.entries()),
   })
+
+  // Process CAR with invocations asynchronously
+  ctx.waitUntil(
```
Comment (alanshaw) [inline packages/access-api/src/routes/raw.js:31]: Should we store only successfully invoked requests? i.e. should we check `res.error`? Can we even? I think no right here.

I assume that later we'll receive a receipt that we can also store so we can know about failures?

cc @Gozala

**Review by alanshaw — CHANGES_REQUESTED** [review]

File: packages/access-api/src/routes/raw.js (hunk 2 above)
Comment (vasco-santos) [reply packages/access-api/src/routes/raw.js:31]: In the UCAN Stream we will see the contents of the `nb` and if it has an error we will skip it. But, if we can skip it here would be better for sure

File: packages/access-api/src/routes/raw.js [hunk 3]
```diff
@@ -17,9 +21,24 @@ export async function postRaw(request, env) {
...
+    (async () => {
+      await fetch(new URL('/ucan', env.uploadApiUrl), {
```
[the remark this answers is not in the document]
Comment (vasco-santos) [reply packages/access-api/src/routes/raw.js:33]: If we fail we won't rollback what is there as I mentioned in PR description... I would like to see this in a Queue but we still need to get access to CF Queue.

We can wrap in retry and block on the user. But given there is no rollback mechanism I believe operations can then fail forever when user retries given that they already succeeded. Thoughts?

meanwhile adding the retry in waitUntil

**Comment by vasco-santos** [comment]
Closing in favour of https://github.com/web3-storage/w3protocol/pull/592

**Pull request closed without merging**